systemd/esm-cache.service

```diff
@@ -13,3 +13,13 @@ Description=Update the local ESM caches
 [Service]
 Type=oneshot
 ExecStart=/usr/bin/python3 /usr/lib/ubuntu-advantage/esm_cache.py
+AppArmorProfile=-ubuntu_pro_esm_cache
+CapabilityBoundingSet=~CAP_SYS_ADMIN
+CapabilityBoundingSet=~CAP_NET_ADMIN
+CapabilityBoundingSet=~CAP_NET_BIND_SERVICE
+CapabilityBoundingSet=~CAP_SYS_PTRACE
+CapabilityBoundingSet=~CAP_NET_RAW
+PrivateTmp=true
+RestrictAddressFamilies=~AF_NETLINK
+RestrictAddressFamilies=~AF_PACKET
+
```

Comment:
This is awesome. If you've got more time and interest, maybe adding additional seccomp filters would be nice, too. I've got these on one of my services, it feels like a reasonable match for what I think these are doing:

```
SystemCallFilter=@basic-io @file-system @io-event @network-io @process @signal @sync @timer brk mprotect ioctl sysinfo uname
```

I can't recall the exact details but missing syscalls are probably logged to `dmesg` or `auditd` as they are encountered and denied.

Reply:
Oh, I was trying with a whole bunch of restrictions before (see here: https://github.com/canonical/ubuntu-pro-client/blob/9e9df67e5a4e3c339caced0d6d19f44bc38799d8/systemd/apt-news.service#L26). But I encountered some difficulties:
- some tests started failing after those changes, and I couldn't figure out why
- unlike apparmor, there are no nice DENIED messages when something was blocked by these :( So I can only use trial and error (enable, run tests, check results), or some deduction. In both cases, it was taking way too long. I plan to return to these.